test/jdk/javax/swing/JComponent/7154030/bug7154030.java

```diff
@@ -86,14 +90,17 @@ public void run() {
 
                     frame.setContentPane(desktop);
                     frame.setSize(300, 300);
-                    frame.setLocation(0, 0);
+                    Dimension screenSize = Toolkit.getDefaultToolkit().getScreenSize();
+                    locx = screenSize.width/2;
+                    locy = screenSize.height/2;
+                    frame.setLocation(locx, locy);
```

Comment:
To move it to the center you can use, setLocationRelativeTo(null) and you will need to get the actual bounds of the window(locx and locy might be ignored/tweaked by the OS)

Reply:
I know setLocationRelativeTo(null) will move to center of screen but I needed the coordinate for robot interaction.
The locx,locy calculation to move to center of screen is done intentionally as we need to use the same coordinate in robot to capture the screen

Reply:
> The locx,locy calculation to move to center of screen is done intentionally as we need to use the same coordinate in robot to capture the screen

You can use locx/loxy intentionally to move the window, but you can't use it in a robot because of:
> you will need to get the actual bounds of the window(locx and locy might be ignored/tweaked by the OS)

Reply:
Modified to use bounds.

Reply:
GC.getBounds() and Toolkit.getScreenSize() return the same bounds, but the bounds of the screen, not the window bounds.

Reply:
I guess we need the screen bounds only, no? I guess by window bounds, u mean frame bounds which is anyway set to 300,300 which I did not change.
If not, can you say which API to use?

Reply:
Used window bounds...Please let me know if any other comments.